apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/common/db/SearchQueryParser.java

```diff
@@ -9,8 +9,12 @@
 import java.util.regex.Pattern;
 
 public class SearchQueryParser {
+    public static final String KEY = "([\\w.]+?)";
+    public static final String OPERATOR = "(=|>|>=|<|<=)";
+    public static final String VALUE = "([\\w:\\-;\\p{Space}]+?)";
+
     public static Collection<SearchCriteria> parse(@NonNull String searchQuery) {
-        Pattern pattern = Pattern.compile("([\\w.]+?)(:|>|>=|<|<=)((\\w|:|-|;|\\p{Space})+?),", Pattern.UNICODE_CHARACTER_CLASS);
+        Pattern pattern = Pattern.compile(KEY + OPERATOR + VALUE + ",", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
As I commented earlier I do not like idea of splitting regex into separate substrings.

Reply:
this can be discussed again and changed later. For the moment, I'll rebase the PR.